pandas/_libs/period.pyx

```diff
@@ -693,7 +693,12 @@ cdef class _Period(object):
 
     def __richcmp__(self, other, op):
         if is_period_object(other):
-            if other.freq != self.freq:
+            try:
+                if other.freq != self.freq:
+                    msg = _DIFFERENT_FREQ.format(self.freqstr, other.freqstr)
+                    raise IncompatibleFrequency(msg)
+            except SystemError:
```

Comment:
I don't think SystemError exists in py2?

Reply:
At least in 2.7.14 it does.  I don't know about earlier off the top.

Reply:
it's there in 2.7.12 and 2.7.9